openshift_cli_installer/libs/clusters/ipi_cluster.py

```diff
@@ -181,3 +182,75 @@ def destroy_cluster(self):
         self.run_installer_command(action=DESTROY_STR, raise_on_failure=True)
         self.logger.success(f"{self.log_prefix}: Cluster destroyed")
         self.delete_cluster_s3_buckets()
+
+
+class AwsIpiCluster(IpiCluster):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.logger = get_logger(f"{self.__class__.__module__}-{self.__class__.__name__}")
+
+        self.platform = AWS_STR
+        if self.create:
+            self._create_install_config_file(platform=self.platform)
+        self.dump_cluster_data_to_file()
+
+
+class GcpIpiCluster(IpiCluster):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.logger = get_logger(f"{self.__class__.__module__}-{self.__class__.__name__}")
+
+        self.platform = GCP_STR
+        self.gcp_sa_file_dir = os.path.join(os.path.expanduser("~"), ".gcp")
+        self.gcp_sa_file_path = os.path.join(self.gcp_sa_file_dir, "osServiceAccount.json")
+        self.tmp_path = os.path.join("/tmp", "installerTemp.json")
+
+        if self.create:
+            self.terraform_parameters.update({"gcp_project_id": self.get_gcp_project_id()})
```

Comment:
1. `self.gcp_project_id` should be defined in the init
2. should be set in `_create_terraform_parameters` - another if in the function